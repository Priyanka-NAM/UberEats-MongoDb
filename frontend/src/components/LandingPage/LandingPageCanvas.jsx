/* eslint-disable react/prefer-stateless-function */
import React, { Component } from "react";
import { Link } from "react-router-dom";
import { BiX } from "react-icons/bi";
import { Button, Row, Container } from "react-bootstrap";
import { OffCanvas, OffCanvasMenu } from "react-offcanvas";
import PropTypes from "prop-types";
import "../Styles/Header.css";

class LandingPageCanvas extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  handleClose = () => {
    const { handleClose } = this.props;
    handleClose();
  };

  render() {
    const { showModal } = this.props;

    return (
      <OffCanvas
        style={{ backgroundColor: "black" }}
        transitionDuration={1000}
        effect='parallax'
        isMenuOpened={showModal}
        position='left'>
        <OffCanvasMenu
          style={{
            backgroundColor: "white",
            width: "15%",
            height: "100%",
            zIndex: "1000",
          }}>
          <Row style={{ marginTop: "5%", marginLeft: "50%" }}>
            <BiX
              size='35px'
              style={{ color: "black" }}
              onClick={this.handleClose}
            />
          </Row>
          <Container align='left' style={{ marginLeft: "15px" }}>
            <Row style={{ marginTop: "20%" }}>
              <Link
                style={{
                  textDecoration: "none",
                  color: "black",
                  fontWeight: "500",
                  fontfamily: "UberMoveText, sans-serif",
                  fontSize: "22px",
                  letterSpacing: "0.05em",
                }}
                to='/owner/signup'>
                Add your Restaurant
              </Link>
            </Row>
            <Row style={{ marginTop: "10%" }}>
              <Link
                style={{
                  textDecoration: "none",
                  color: "black",
                  fontWeight: "500",
                  fontfamily: "UberMoveText, sans-serif",
                  fontSize: "22px",
                  letterSpacing: "0.05em",
                }}
                to='/customer/signup'>
                Sign up to order
              </Link>
            </Row>
            <Row style={{ marginTop: "15%" }}>
              <Link to='/signin'>
                <Button
                  variant='dark'
                  style={{
                    width: "60%",
                    height: "60px",
                    fontWeight: "500",
                    fontfamily: "UberMoveText, sans-serif",
                    fontSize: "22px",
                    letterSpacing: "0.05em",
                  }}
                  onClick={this.handleClose}>
                  Sign In
                </Button>
              </Link>
            </Row>
          </Container>
        </OffCanvasMenu>
      </OffCanvas>
    );
  }
}
LandingPageCanvas.propTypes = {
  showModal: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
};
export default LandingPageCanvas;
