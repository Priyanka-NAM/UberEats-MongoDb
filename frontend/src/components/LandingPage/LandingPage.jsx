/* eslint-disable import/no-unresolved */
import React, { Component } from "react";
import { Nav, Button, Row } from "react-bootstrap";
import { FaBars } from "react-icons/fa";
import { Redirect } from "react-router";
import { Link } from "react-router-dom";

import UberELogo from "../Home/HomeIcons/logo";
import mainstyle from "../Home/HomeIcons/HeaderStyle";
import LandingPageCanvas from "./LandingPageCanvas";
import "../Styles/Header.css";

class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
    };
  }

  componentDidMount() {
    document.body.style.backgroundSize = "2200px 1200px";
    document.body.style.backgroundRepeat = "no-repeat";
    document.body.style.overflow = "hidden";
    document.body.style.backgroundImage = `url("/images/Startpage_bg.PNG")`;
  }

  componentWillUnmount() {
    document.body.style.backgroundImage = ``;
  }

  handleShow = () => {
    this.setState({
      showModal: true,
    });
  };

  handleClose = () => {
    this.setState({
      showModal: false,
    });
  };

  render() {
    const { showModal } = this.state;
    let redirectVar = null;
    const user = JSON.parse(localStorage.getItem("user"));
    if (user && user.is_owner === 0) {
      redirectVar = <Redirect to='/customer/home' />;
    } else if (user && user.is_owner === 1) {
      redirectVar = <Redirect to='/owner/home' />;
    }

    return (
      <div className='imgStyle'>
        {redirectVar}
        <Row fluid='true'>
          <Nav style={mainstyle.headerRow}>
            <Nav.Item style={{ paddingLeft: "50px" }}>
              <FaBars
                xs='6'
                size='22px'
                color='black'
                onClick={this.handleShow}
              />
            </Nav.Item>
            <Nav.Item style={{ paddingLeft: "20px" }}>
              <img
                style={mainstyle.paddingLeft}
                src={UberELogo.UberEBLogo.src}
                alt={UberELogo.UberEBLogo.alt}
              />
            </Nav.Item>
            <Nav.Item style={{ paddingLeft: "70%" }}>
              <Link to='/signin'>
                <Button
                  style={{
                    fontSize: "20px",
                    fontWeight: "500",
                    width: "150%",
                    backgroundColor: "white",
                    height: "60px",
                    borderRadius: "40px",
                    fontFamily: "UberMoveText, sans-serif",
                  }}
                  variant='light'>
                  Sign in
                </Button>
              </Link>
            </Nav.Item>
          </Nav>
        </Row>
        <LandingPageCanvas
          handleClose={this.handleClose}
          showModal={showModal}
        />
      </div>
    );
  }
}

export default LandingPage;
