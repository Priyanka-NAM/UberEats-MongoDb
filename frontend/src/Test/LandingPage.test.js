/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";

import Adapter from "enzyme-adapter-react-16";

import LandingPage from "../components/LandingPage/LandingPage";

Enzyme.configure({ adapter: new Adapter() });

it("render correctly the Landing Page component", () => {
  const LandingPageComponent = renderer
    .create(
      <Router>
        <LandingPage />
      </Router>
    )
    .toJSON();
  expect(LandingPageComponent).toMatchSnapshot();
});
