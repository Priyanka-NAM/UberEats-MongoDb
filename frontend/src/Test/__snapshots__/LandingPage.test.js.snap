// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the Landing Page component 1`] = `
<div
  className="imgStyle"
>
  <div
    className="row"
    fluid="true"
  >
    <div
      className="nav"
      onKeyDown={[Function]}
      style={
        Object {
          "marginLeft": "2rem",
          "marginTop": "2rem",
          "width": "100%",
        }
      }
    >
      <div
        className="nav-item"
        style={
          Object {
            "paddingLeft": "50px",
          }
        }
      >
        <svg
          color="black"
          fill="currentColor"
          height="22px"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "black",
            }
          }
          viewBox="0 0 448 512"
          width="22px"
          xmlns="http://www.w3.org/2000/svg"
          xs="6"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <div
        className="nav-item"
        style={
          Object {
            "paddingLeft": "20px",
          }
        }
      >
        <img
          alt="Uber Eats Home"
          src="https://d3i4yxtzktqr9n.cloudfront.net/web-eats-v2/ee037401cb5d31b23cf780808ee4ec1f.svg"
          style={
            Object {
              "paddingLeft": "1rem",
            }
          }
        />
      </div>
      <div
        className="nav-item"
        style={
          Object {
            "paddingLeft": "70%",
          }
        }
      >
        <a
          href="/signin"
          onClick={[Function]}
        >
          <button
            className="btn btn-light"
            disabled={false}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": "40px",
                "fontFamily": "UberMoveText, sans-serif",
                "fontSize": "20px",
                "fontWeight": "500",
                "height": "60px",
                "width": "150%",
              }
            }
            type="button"
          >
            Sign in
          </button>
        </a>
      </div>
    </div>
  </div>
  <div>
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "backgroundColor": "white",
          "height": "100%",
          "left": "-250px",
          "position": "fixed",
          "right": "auto",
          "top": "0px",
          "transform": "translate(-250px, 0px)",
          "transitionDuration": "1000ms",
          "width": "15%",
          "zIndex": "1000",
        }
      }
    >
      <div
        className="row"
        style={
          Object {
            "marginLeft": "50%",
            "marginTop": "5%",
          }
        }
      >
        <svg
          fill="currentColor"
          height="35px"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "black",
            }
          }
          viewBox="0 0 24 24"
          width="35px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.192 6.344L11.949 10.586 7.707 6.344 6.293 7.758 10.535 12 6.293 16.242 7.707 17.656 11.949 13.414 16.192 17.656 17.606 16.242 13.364 12 17.606 7.758z"
          />
        </svg>
      </div>
      <div
        align="left"
        className="container"
        style={
          Object {
            "marginLeft": "15px",
          }
        }
      >
        <div
          className="row"
          style={
            Object {
              "marginTop": "20%",
            }
          }
        >
          <a
            href="/owner/signup"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "fontSize": "22px",
                "fontWeight": "500",
                "fontfamily": "UberMoveText, sans-serif",
                "letterSpacing": "0.05em",
                "textDecoration": "none",
              }
            }
          >
            Add your Restaurant
          </a>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginTop": "10%",
            }
          }
        >
          <a
            href="/customer/signup"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "fontSize": "22px",
                "fontWeight": "500",
                "fontfamily": "UberMoveText, sans-serif",
                "letterSpacing": "0.05em",
                "textDecoration": "none",
              }
            }
          >
            Sign up to order
          </a>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginTop": "15%",
            }
          }
        >
          <a
            href="/signin"
            onClick={[Function]}
          >
            <button
              className="btn btn-dark"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "fontSize": "22px",
                  "fontWeight": "500",
                  "fontfamily": "UberMoveText, sans-serif",
                  "height": "60px",
                  "letterSpacing": "0.05em",
                  "width": "60%",
                }
              }
              type="button"
            >
              Sign In
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
